### ISSUE TITLE
Incomplete output displayed and [Done] exited with code=null

### ISSUE THREAD

**Issue by sulbig** [formulahendry/vscode-code-runner#171]
When running a program with lots of console output, I'm getting a "[Done] exited with code=null in 0.0XX seconds" before all of the output has actually been written to the console.

Below is a simple Python3 example to reproduce the problem. Each time the code is run, the console output will enumerate to a different count value before it stops with the code=null exit code.

```
for count in range(100000):
    print(count)
```

The output window displays...

```
...
37345
37346
37347
37348
37349
37
[Done] exited with code=null in 0.044 seconds
```

If I run the same example in the Visual Studio Code terminal window, the full expected output is displayed every time.

**Comment by formulahendry** [comment]
For quick turnaround, you could use below setting (`File`->`Preference`->`Settings`) to run code in Integrated Terminal:
```json
{
    "code-runner.runInTerminal": true
}
```
Also, you may try below setting to run:
```json
{
    "code-runner.executorMap.python": "python -u"
}
```

Are those options working for you?

**Comment by yinguobing** [comment]
Same issue, and the first option works.
```
{
    "code-runner.runInTerminal": true
}
```

**Comment by sulbig** [comment]
The first option suggested successfully displays every print statement in the example above in the Terminal window, every time.
```
// First option
{
    "code-runner.runInTerminal": true
}
```


The second option suggested has the original issue displayed in the Output window.
```
// Second option
{
    "code-runner.executorMap.python": "python -u"
}
```

**Comment by sulbig** [comment]
I'm not a Javascript/NodeJS programmer, but I did some reading of the code in this extension, as well as the NodeJS API, and think I may have come across something interesting.

I see that the **exec()** fucntion of child_process is being used in the executeCommandInOutputChannel() function in the codeManager.ts file. I'm understanding that exec() spawns a shell and executes the command within that shell, buffering any generated output. There is a maxBuffer option that is documented as:

> maxBuffer \<number\> : Largest amount of data in bytes allowed on stdout or stderr. (Default: 200*1024) If exceeded, the child process is terminated. See caveat at maxBuffer and Unicode.

[https://nodejs.org/api/child_process.html#child_process_child_process_exec_command_options_callback](https://nodejs.org/api/child_process.html#child_process_child_process_exec_command_options_callback)

It looks as though the **spawn()** function should be used instead, since the data is streamed as standard IO objects -- not buffered strings.

**Comment by Kogam22** [comment]
```
For quick turnaround, you could use below setting (File->Preference->Settings) to run code in Integrated Terminal:

{
    "code-runner.runInTerminal": true
}
Also, you may try below setting to run:

{
    "code-runner.executorMap.python": "python -u"
}
Are those options working for you?
```

@formulahendry 

Thanks Man. 
BTW, What's The Actual Problem Here ?

**Comment by Khalilsqu** [comment]
Same problem python vsc windows 10

**Comment by z33kz33k** [comment]
This issue can lead to pretty nasty problems (debugging-wise) if one decides to e.g. create some output files and print some lengthy message about it on stout in the same loop. What happens is, SILENTLY, some of the files will go missing. Consider the below Python example:

```python
for i in range(1, 4501):
        prefix = "debug/vscodetest/"
        suffix = ".txt"
        path = "{}{}{}".format(prefix, str(i), suffix)
        with open(path, mode="w") as f:
            f.write("Hello world!")
            
        print("#{} file printed {}".format(str(i), "padding"*3))
```
It only takes to change `padding*3` to `padding*4` to have couple hundred files gone missing. Have a good time noticing it, let alone trying to figure out what happened (especially considering these rather aren't gonna be some helloworlds in a simple range() loop). 

In my case I first thought it was some strange problem with Python (performance issue, stout redirection I used not working properly etc.), then that it probably is some strange problem with VS Code and at last after one day at work spent on this I finally landed here.

I think the problem at least merits some more visibility (maybe its own separate issue?)

**Comment by nbBIM** [comment]
The same have happened to me, but in my case after applying your suggestions (_specially the one of running Python script in the Terminal_), I can see the starting point of the output

**Comment by oleghwang** [comment]
Hello, can anybody help me please. When i'm running terminal, i 've got very long details in terminal. How can i remove it, just to have the result of code
thank you in advance  
![terminal](https://user-images.githubusercontent.com/48288911/53891599-15554200-406e-11e9-99e3-0138d8de3b65.png)

**Comment by oleghwang** [comment]
![terminal2](https://user-images.githubusercontent.com/48288911/53891944-b5ab6680-406e-11e9-967d-69884c6ef674.png)

**Comment by nuptownboy** [comment]
Did Oleg Hwang get an answer to this question as I have same question exactly and can only find the question and no answer?

**Comment by sauravexodus** [comment]
If you run the code in the terminal then it works. Usually, it is a buffer issue and your terminal will have a larger buffer.

**Comment by JedHmr** [comment]
First option works

> For quick turnaround, you could use below setting (`File`->`Preference`->`Settings`) to run code in Integrated Terminal:
> 
> ```json
> {
>     "code-runner.runInTerminal": true
> }
> ```
> 
> Also, you may try below setting to run:
> 
> ```json
> {
>     "code-runner.executorMap.python": "python -u"
> }
> ```
> 
> Are those options working for you?

Works for VS code running python; note further that the output window was not printing in a 'co-moving' fashion so I couldn't see the output as it went along. Thanks.

**Comment by JeffreyCA** [comment]
@formulahendry could you take a look at https://github.com/formulahendry/vscode-code-runner/pull/428?

**Comment by formulahendry** [comment]
Fixed in v0.9.13.  Thanks @JeffreyCA !!
